release/cmd/hashrelease.go

```diff
@@ -231,6 +233,13 @@ func hashreleaseSubCommands(cfg *Config) []*cli.Command {
 				if reg := c.StringSlice(registryFlag.Name); len(reg) > 0 {
 					opts = append(opts, calico.WithImageRegistries(reg))
 				}
+				if !c.Bool(publishHashreleaseImageFlag.Name) {
```

Comment:
Shouldn't this be `if c.Bool` instead of `if !c.Bool`?

Reply:
The list of components is needed when not publishing images built by the hashrelease locally.

Reply:
Let's add a comment saying that then - this is super unintuitive (and caught both me and @coutinhop on this one!)

```
// We only need to check that the correct images exist if we haven't built them ourselves. So, skip this check
// if we're configured to build and publish images from the local codebase.
```